### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by Pradhvan**
* Added PR template with a checklist for the author.
* Added instructions in the checklists that points to the wiki.
* Added a heading to reference PR template to related issue it closes.

Closes https://github.com/scanapi/scanapi/issues/383

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## Checklist
+<!-- If any particular item isn't necessary with your change, check it anyway so that the reviewer knows nothing is pending in the PR --> 
+
+- [ ] Added a changelog entry
+- [ ] Added unit tests
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] I have self-documented code my changes by adding docstring(s) and comment(s).
+- [ ] Current PR does not significantly decrease the code coverage and docstring coverage.
+- [ ] My code follows the style guidelines of this project.
```
Comment (camilamaia) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:18]:
```suggestion
- [ ] My code follows the style guidelines of this project.

## Issue
<!--- All PRs must have a related issue. This way we can ensure that no one loses time working in something that does not needed to be done. -->

Closes <issue_number>
```

**Review by camilamaia — COMMENTED** [review]
Thanks @Pradhvan! I've added some suggestions, let me know what do you think :)

**Comment by Pradhvan** [comment]
> Thanks @Pradhvan! I've added some suggestions, let me know what do you think :)

Changes look good to me. I will update the PR. 👍🏾

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (Pradhvan) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:18]: @camilamaia added this part in the template but we might have to remove this when we finish the commit pattern design. As ideally, you should add a `Closes <issue link>` at the end of the commit message. 

That automatically references the related issue. But let's see how that goes when we finalize the commit pattern. Let's work with this for now. 😄

Comment (camilamaia) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:18]: @Pradhvan the same way works for the PRs! When closing the PR, it closes the issue too. 

I don't mind having this info in both places. It is easier for reviewers when the info is in the PR. But it is easy for checking history when the info is in the commit. What do you think?

**Review by camilamaia — APPROVED** [review]
⭐

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (Pradhvan) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:18]: Sounds good to me. 👍🏾

**Pull request merged**